app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[self.name] = self
+
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    result = [Person(person["name"], person["age"]) for person in people]
+    for person_dict in people:
+        person = Person.people[person_dict["name"]]
+        if person_dict.get("wife"):
+            person.wife = Person.people[person_dict["wife"]]
+        elif person_dict.get("husband"):
+            person.husband = Person.people[person_dict["husband"]]
```

Comment:
if you so much times point to Person.people, consider creating local reference to the Person.people dictionary in your function, like person_people, so we won't every time search this link inside Python, which will make it more concise and will a little improve perfomance.